### ISSUE TITLE
Settings > "Edit" link lacks context (Accessibility)

### ISSUE THREAD

**Issue by tofumatt** [google/site-kit-wp#6642]
## Bug Description

The “Edit” link for each section in Settings (Search Console, Analytics, etc.) lacks context and should have some visually hidden text so it reads “Edit Search Console settings” instead of just “Edit”.

## Steps to reproduce

1. Go to Site Kit Settings page
2. Expand any module
3. Navigate to the "Edit" link for the module using a screen reader
4. See a basic "Edit" link with little context

## Screenshots

https://user-images.githubusercontent.com/90871/221317460-f4aff0c4-3e3f-4bc8-8188-a9bf3c438a30.mp4

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The “Edit” link for a module should be updated to `“Edit<VisuallyHidden> [MODULE NAME] settings</VisuallyHidden>”`.

## Implementation Brief

* In `assets/js/components/settings/SettingsActiveModule/Footer.js`
  * Wrap the existing `Edit` string with a `span` and give it a `aria-hidden` so the text is hidden for screen readers.
    https://github.com/google/site-kit-wp/blob/cac2239c196261c90d2e9d496e4f54cc151e5763/assets/js/components/settings/SettingsActiveModule/Footer.js#L210-L221
  * Next to the `<span>` add a `<VisuallyHidden>` Element that contains the output for screen readers: "Edit [MODULE NAME] settings"
    * Use `sprintf` to inject the module name into the translation string, see
      https://github.com/google/site-kit-wp/blob/cac2239c196261c90d2e9d496e4f54cc151e5763/assets/js/components/settings/SettingsActiveModule/Footer.js#L232-L236

### Test Coverage

* Verify VRT tests don't fail
* Verify the screen reader reads "Edit MODULE NAME settings"

## QA Brief

* Navigate to the Site Kit settings with a screenreader (like macOS's VoiceOver) enabled.
* Tab to the "Edit" link in the footer of a module in the list of modules.
* The label read should be "Edit [MODULE NAME] settings", not "Edit".

## Changelog entry

* Add an aria-label to the edit link on the settings view component.

**Comment by eugene-manuilov** [comment]
> I deviated from the AC that recommeded to use `“Edit<VisuallyHidden> [MODULE NAME] settings</VisuallyHidden>”`.
> 
> This solution only works under the assumption that chaining the string in this order makes sense for all languages. That's not the case, e.g. for German you would put the "Edit" part at the end of the string.
> 
> Therefore I recommend using 2 different strings / solutions for visual UI an screen readers.

@derweili I think the idea in AC was to use the `createInterpolateElement` function to generate one translatable string as we do it in other places like here:

https://github.com/google/site-kit-wp/blob/a3fa8f3dd0e23c49b34e7ed64b11441637e27a3e/assets/js/modules/tagmanager/components/settings/SettingsView.js#L121-L143

**Comment by eugene-manuilov** [comment]
Ah... okay, thanks, @derweili, that makes sense. Ok, then IB looks good to me and i'll move it to EB. Also, looks like we need to rework our usage of visual hidden element and `createInterpolateElement` function to produce correct translations for all languages. @tofumatt what do you think?

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:**

- With a screenreader enabled. The label reads "Edit [MODULE NAME] settings", not "Edit".
- I tested this by using macOS's VoiceOver.  See screencast.  
- In addition, I can see in the aria-label settings are correct too. 
- Checked all modules within Site Kit. 

<details>
 <summary>Details</summary>

![image](https://github.com/google/site-kit-wp/assets/73545194/3261992c-4cc0-472c-8010-7681e35189c1)
![image](https://github.com/google/site-kit-wp/assets/73545194/366ab88c-dd43-4f01-af81-09d7bb428991)

https://github.com/google/site-kit-wp/assets/73545194/37c08de4-32fd-4b1c-9b21-482ece4a6ce3

</details>
